instruments/cryomagnetics/cryomagnetics_lm510.py

```diff
@@ -0,0 +1,193 @@
+    querying_lock = Lock()  # type: Lock
+
+    measurement_timeout = 1
+
+    UNITS = {
+        "cm": pq.cm,
+        "in": pq.inch,
+        "%": pq.percent,
+        "percent": pq.percent
+    }
+
+    @property
+    def default_channel(self):
+        """
+        The Level meter has a default channel that is used to measure
+        different cryogen levels. If a channel is not specified for a
+        particular command, the command gets sent to the value of the
+        default channel instead
+
+        :return: The current default channel
+        :rtype: int
+        """
+        return int(self.query("CHAN?"))
+
+    @default_channel.setter
+    def default_channel(self, channel):
+        """
+        Set the default channel to the desired channel
+
+        :param int channel: The desired channel
+        :raises: :exc:`ValueError` if the channel to be set is not in the
+        set of allowed channels defined in ``CHANNELS``
+        """
+        if channel not in self.CHANNELS:
+            raise ValueError("Attempted to set channel to %s. Channel must "
+                             "be an integer of either 1 or 2", channel)
+        command = "CHAN %d" % channel
+        self.query(command)
+
+    @property
+    def status_byte(self):
+        """
+        Get a byte array representing the instrument's status byte. The
+        status byte describes whether channel 1 or 2 are ready to make a
+        measurement. The structure of the status byte is described in
+        greater detail in the level meter's operating manual.
+
+        :return: The current value of the status byte
+        :rtype: bytes
+        """
+        byte_as_string = self.query("*STB?")
+        return byte_as_string.encode('ascii')
+
+    @property
+    def channel_1_data_ready(self):
+        """
+
+        :return: ``True`` if channel 1 is ready to measure, otherwise ``False``
+        :rtype: bool
+        """
+        status_byte = self.status_byte
+        return (int(status_byte[0]) & 1) > 0
+
+    @property
+    def channel_2_data_ready(self):
+        """
+
+        :return: ``True`` if channel 2 is ready to measure, otherwise ``False``
+        :rtype: bool
+        """
+        status_byte = self.status_byte
+        return (int(status_byte[0]) & 4) > 0
+
+    @property
+    def channel_1_measurement(self):
+        """
+        Make a measurement on channel 1
+
+        :return: The level of the cryogen.
+        :rtype: Quantity
+        """
+        return self._measurement(1)
+
+    @property
+    def channel_2_measurement(self):
+        """
+        Make a measurement on channel 2
+
+        :return: The level of cryogen measured by the probe connected to
+            channel 2 in the meter
+        :rtype: Quantity
+        """
+        return self._measurement(2)
+
+    def reset(self):
+        """
+        Bring the instrument to a safe, known state
+        """
+        self.query("*RST")
```

Comment:
Is this using ``query()`` instead of ``sendcmd()`` due to the echoing behavior handled by the communicator above? If so, I'd suggest explicitly noting such, perhaps in a comment, since ``query()`` can fail horribly if there's no data returned— a naïve maintainer may later mistake this line for an error and "fix" it to use ``sendcmd()``.